code/lib/preview-api/src/modules/store/csf/prepareStory.ts

```diff
@@ -61,12 +58,11 @@ export function prepareStory<TRenderer extends Renderer>(
       normalizeArrays(componentAnnotations.loaders),
       normalizeArrays(storyAnnotations.loaders),
     ]) {
-      const loadResults = await Promise.all(loaders.map((loader) => loader(updatedContext)));
-      const loaded: Record<string, any> = Object.assign({}, ...loadResults);
-      updatedContext = { ...updatedContext, loaded: { ...updatedContext.loaded, ...loaded } };
+      if (context.abortSignal.aborted) return loaded;
```

Comment:
aborting as  early as possible inside of applyloaders

Reply:
How is this scenario possible? Is there an `await` in between the `StoryRendering` checking the abort signal and this line?

Reply:
Yes the line below this line:
`const loadResults = await Promise.all(loaders.map((loader) => loader(context)));`
And notice that this is a loop, calling the project loaders first, and then the component and then the story loaders.

Whenever the user navigates to a new story, or force remounts, the StoryRender file will call:
`this.abortController?.abort();`

Now, the loaders can be cancelled half way, before it would only cancel after all loader are loaded.

Reply:
Ahh OK, got it. Maybe a comment would be useful here (something like: "If an abort is received in between running each level of loaders, we want to avoid running the next level")

Reply:
Okay, will add that to the mount PR!